tests/testsuite.src/run_misc.at

```diff
@@ -15251,3 +15289,358 @@ AT_CHECK([$COMPILE prog.cob])
+           ELSE
+             DISPLAY "HIGH-VALUE KO"
+           END-IF.
+           IF "X" > LOW-VALUE AND "X" > LV
+              DISPLAY "X > LOW-VALUE"
+           ELSE
+              DISPLAY "X < LOW-VALUE"
+           END-IF.
+           IF "X" < HIGH-VALUE AND "X" < HV
+              DISPLAY "X < HIGH-VALUE"
+           ELSE
+              DISPLAY "X > HIGH-VALUE"
+           END-IF.
+           CALL "prog2".
+           STOP RUN.
+       END PROGRAM prog1.
+       IDENTIFICATION DIVISION.
+       PROGRAM-ID. prog2.
+       ENVIRONMENT DIVISION.
+       CONFIGURATION SECTION.
+       OBJECT-COMPUTER. x86,
+           PROGRAM COLLATING SEQUENCE IS alpha-ascii.
+       SPECIAL-NAMES.
+           ALPHABET alpha-ascii IS ASCII.
+       DATA DIVISION.
+       WORKING-STORAGE SECTION.
+       01 LV     PIC X.
+       88 IS-LV  VALUE LOW-VALUE.
+       01 LV-VAL REDEFINES LV BINARY-CHAR UNSIGNED.
+       01 HV     PIC X.
+       88 IS-HV  VALUE HIGH-VALUE.
+       01 HV-VAL REDEFINES HV BINARY-CHAR UNSIGNED.
+       PROCEDURE DIVISION.
+           MOVE LOW-VALUE TO LV.
+           MOVE HIGH-VALUE TO HV.
+           DISPLAY "P2 LOW-VALUE: " LV-VAL
+                 " HIGH-VALUE: " HV-VAL.
+           IF LV = LOW-VALUE AND IS-LV
+             DISPLAY "LOW-VALUE OK"
+           ELSE
+             DISPLAY "LOW-VALUE KO"
+           END-IF.
+           IF HV = HIGH-VALUE AND IS-HV
+             DISPLAY "HIGH-VALUE OK"
+           ELSE
+             DISPLAY "HIGH-VALUE KO"
+           END-IF.
+           IF "X" > LOW-VALUE AND "X" > LV
+              DISPLAY "X > LOW-VALUE"
+           ELSE
+              DISPLAY "X < LOW-VALUE"
+           END-IF.
+           IF "X" < HIGH-VALUE AND "X" < HV
+              DISPLAY "X < HIGH-VALUE"
+           ELSE
+              DISPLAY "X > HIGH-VALUE"
+           END-IF.
+           STOP RUN.
+       END PROGRAM prog2.
+])
+
+AT_CHECK([$COMPILE -febcdic-table=ebcdic500_latin1 prog5.cob], [0], [], [])
+AT_CHECK([$COBCRUN_DIRECT ./prog5], [0],
+[P1 LOW-VALUE: 000 HIGH-VALUE: 159
+LOW-VALUE OK
+HIGH-VALUE OK
+X > LOW-VALUE
+X < HIGH-VALUE
+P2 LOW-VALUE: 000 HIGH-VALUE: 255
+LOW-VALUE OK
+HIGH-VALUE OK
+X > LOW-VALUE
+X < HIGH-VALUE
+], [])
+
+AT_DATA([prog6.cob], [
+       IDENTIFICATION DIVISION.
+       PROGRAM-ID. prog.
+       DATA DIVISION.
+       WORKING-STORAGE SECTION.
+       01 IND PIC 9(02).
+       01 ENDX PIC X(01).
+       01 ENDXS PIC X(05).
+       PROCEDURE DIVISION.
+           INITIALIZE ENDX.
+           PERFORM VARYING IND FROM 1 BY 1 UNTIL ENDX = HIGH-VALUE
+              DISPLAY IND " " WITH NO ADVANCING
+              IF IND = 9
+                 MOVE HIGH-VALUE TO ENDX
+              END-IF
+           END-PERFORM.
+           INITIALIZE ENDXS.
+           PERFORM VARYING IND FROM 1 BY 1 UNTIL ENDXS = HIGH-VALUE
+              DISPLAY IND " " WITH NO ADVANCING
+              IF IND = 9
+                 MOVE HIGH-VALUE TO ENDXS
+              END-IF
+           END-PERFORM.
```

Comment:
as I've missed that - please add a COBOL comment why we do this second loop

Reply:
This was too long ago, I do not remember why. The only thing I think to remember about this test is because our client reported failure in one of the two cases (or both ?).

Reply:
Do you see a possible reason that this _could_ happen with our current codegen? If not, then I tend to drop the second iteration (if you don't want that feel free to add your note as a COBOL comment instead)

Reply:
Checking the generated code in both case, there are no differences except the target field and its length, so yeah I believe the second loop is useless.